src/PointwiseFunctions/Hydro/EquationsOfState/HybridEos.cpp

```diff
@@ -0,0 +1,153 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Hydro/EquationsOfState/HybridEos.hpp"
+
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace EquationsOfState {
+template <typename ColdEquationOfState>
+HybridEos<ColdEquationOfState>::HybridEos(
+    ColdEquationOfState cold_eos, const double thermal_adiabatic_index) noexcept
+    : cold_eos_(std::move(cold_eos)),
+      thermal_adiabatic_index_(thermal_adiabatic_index) {}
+
+EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <typename ColdEquationOfState>,
+                                     HybridEos<ColdEquationOfState>, double, 2)
+EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <typename ColdEquationOfState>,
+                                     HybridEos<ColdEquationOfState>, DataVector,
+                                     2)
+
+template <typename ColdEquationOfState>
+HybridEos<ColdEquationOfState>::HybridEos(CkMigrateMessage*
+                                          /*unused*/) noexcept {}
+
+template <typename ColdEquationOfState>
+void HybridEos<ColdEquationOfState>::pup(PUP::er& p) noexcept {
+  EquationOfState<is_relativistic, 2>::pup(p);
+  p | cold_eos_;
+  p | thermal_adiabatic_index_;
+}
+
+template <typename ColdEquationOfState>
+template <class DataType>
+Scalar<DataType>
+HybridEos<ColdEquationOfState>::pressure_from_density_and_energy_impl(
```

Comment:
@ffoucart would you be willing to check the math in these functions in the cpp file? I've gone through the rest of the code but don't feel super qualified to check the math here :)

Reply:
Sure, I'll have a look. This is a prereq to the SpectralEOS PR that I'm supposed to take over, so that'll be a good exercise for me anyway.

Reply:
Okay, great! Thank you so much :)